src/session/key.ts

```diff
@@ -0,0 +1,106 @@
+import { ethers } from 'ethers'
+import { CONTRACT_ABIS, CONTRACT_ADDRESSES, getFilecoinNetworkType } from '../utils/index.ts'
+
+export const CREATE_DATA_SET_TYPEHASH = ethers.id(
```

Comment:
I think we could lift up EIP712_TYPES in pdp/auth.ts into types.ts and reconstruct these strings using that data so we don't have to repeat ourselves?

Reply:
I don't think there's an ethers util to make typehash from eth_signTypedData_v4 types. I can move the code though.